src/main/java/org/broadinstitute/dsm/route/ClinicalKitsRoute.java

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.dsm.route;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.model.ClinicalKit;
+import org.broadinstitute.dsm.model.ParticipantWrapper;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.util.NotificationUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import java.sql.Connection;
+import java.util.Map;
+import java.util.Optional;
+
+public class ClinicalKitsRoute implements Route {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClinicalKitsRoute.class);
+
+    private NotificationUtil notificationUtil;
+
+    public ClinicalKitsRoute(@NonNull NotificationUtil notificationUtil) {
+        this.notificationUtil = notificationUtil;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+        String kitLabel = request.params(RequestParameter.LABEL);
+        if (StringUtils.isBlank(kitLabel)) {
+            throw new RuntimeException("Please include a kit label as a path parameter");
+        }
+
+        return TransactionWrapper.inTransaction(conn -> {
+            return getClinicalKit(conn, kitLabel, response);
+        });
+    }
+
+    private Object getClinicalKit(Connection conn, String kitLabel, Response response) {
+        logger.info("Checking label " + kitLabel);
+
+        // this method already sets the received time, check for exited and deactivation and special behaviour, and triggers DDP, we don't need a new
+        BSPKitInfo kitInfo = (BSPKitInfo) BSPKitQueryRoute.getBSPKitInfo(conn, kitLabel, response, notificationUtil);
```

Comment:
don't use another route in a route! move shared methods into a shared class (util class or other)

Reply:
I would add here that, since you use some logic from `BSPKitQueryRoute` you could introduce new class for example `BSPKitQuery` and move all necessary logic there and you could freely use object of `BSPKitQuery` in both route `ClinicalKitsRoute` and `BSPKitQueryRoute`.